CMakeLists.txt

```diff
@@ -114,7 +138,7 @@ if (UNIT_TESTING)
   add_subdirectory(sdk/tests/iot/provisioning)
 
   # Platform
-  if(AZ_PLATFORM_IMPL STREQUAL "POSIX")
+  if(AZ_PLATFORM_IMPL STREQUAL "POSIX" AND AZ_MQTT_TRANSPORT_IMPL STREQUAL "MOSQUITTO") # Temporary
```

Comment:
Nit: I wouldn't add the "Temporary" comment. 
These tests do not exist today, and the plan of record is to add integration testing _only_ for Posix/MosquittoMQTT client.

Reply:
Got it, removed the temporary, I had set it there more for our reference once we merge with the main branch (in case we were to test for other mqtt clients). Given that we're **only** doing Mosquitto I'll remove the temporary.